tests/by-util/test_id.rs

```diff
@@ -167,3 +206,32 @@ fn test_id_password_style() {
 
     assert!(result.stdout_str().starts_with(&username));
 }
+
+#[cfg(any(target_vendor = "apple", target_os = "linux"))]
```

Comment:
Is this not possible for unix systems other than Linux and Mac? Same question for the other function.

Reply:
I find it extremely convenient to directly test against GNU coreutils 8.32.
`cmd_keepenv(util_name)` runs whatever util is in `$PATH` on the system the test runs on.
GNU coreutils is installed on the docker ubuntu images that github actions is using and I patched github actions to also install it for the macOS tests.
On every other system I can't guarantee that the util version is GNU. (If it's not GNU the tests may differ, you experienced this before when you said you had shadow coreutils installed).

Reply:
I agree that it's very convenient. However, I see a couple of issues with this:
- Other systems don't get tested as much and we might overlook subtle differences. The shadow-utils groups command also comes with Manjaro (and probably other Arch-based distros) by default, so the tests don't work there.
- It becomes circular when people run the tests on their machine on which they have uutils installed instead of GNU.


I understand that sometimes it's our best option, but I'd prefer to use it as little as possible.

Reply:
All true.

However, there's not really an alternative for tests with dynamically expected results (except quirky work-arounds). Best example is `id --groups`. Before I changed the tests, it was only tested if the stdout could be converted into numbers (fun fact, the parsing target was **f**64).

Imho. it's not a priority to have tests that support every linux distro, but it's of course important to document (`CONTRIBUTING.md`?) that some tests rely on GNU coreutils in the `$PATH`.
 
It's probably a good idea to implement a "feature" to tag the tests that rely on having GNU coreutils in the `$PATH`. So that it's easily possible to switch these tests off when running all tests in an environment without this prerequisite.

Reply:
I like the feature flag idea!